analysis/node-wrapper/install.js

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env node
+
+const fs = require("fs");
+const path = require("path");
+var isWindows = require("is-windows");
+
+if (isWindows() === true && process.env.npm_package_name === "codecharta-analysis") {
+  (async () => {
+      const bins = ["ccsh.ps1", "ccsh.cmd"];
+      const globalOrLocal = process.env.npm_config_global === "true";
+      const scriptLocation = globalOrLocal ? "binScripts/global" : "binScripts/local";
+      const scriptTarget = globalOrLocal
+        ? process.env.npm_config_global_prefix
+        : path.join(process.env.npm_config_local_prefix, "node_modules/.bin");
+
+      await Promise.all(bins.map((async (file) => {
+        try {
+          await fs.promises.copyFile(path.join(scriptLocation, file), path.join(scriptTarget, file));
+          console.log(`${file} was moved to ${scriptTarget}`);
+        } catch (err) {
+          console.error("File could not get copied")
+          throw err
+        }
```

Comment:
```suggestion
        await fs.promises.copyFile(path.join(scriptLocation, file), path.join(scriptTarget, file));
        console.log(`${file} was moved to ${scriptTarget}`);
```

Using try / catch like this is an anti-pattern. The error itself already contains the necessary information and it will show up when the code fails. Please be aware that the script is going to fail as soon as a single error is coming up.

Reply:
Just one more thing:
Why `await` the file copy? Doesn't this defeat the whole async  point? Isn't the `await Promise.all([])` enough to wait for all results?

Reply:
That depends, if you want to log that a file was moved or not. It's possible to directly return in case you don't want to log it.